src/jdk.crypto.ec/share/classes/sun/security/ec/ECOperations.java

```diff
@@ -486,5 +413,286 @@ public boolean checkOrder(ECPoint point) {
+                    lookup(pointMultiples, low, lookupResult);
+                    ecOps.setSum(result, lookupResult, t0, t1, t2, t3, t4);
+                }
+
+                return result;
+            }
+
+            private void double4(ProjectivePoint.Mutable p,
+                    MutableIntegerModuloP t0, MutableIntegerModuloP t1,
+                    MutableIntegerModuloP t2, MutableIntegerModuloP t3,
+                    MutableIntegerModuloP t4) {
+                for (int i = 0; i < 4; i++) {
+                    ecOps.setDouble(p, t0, t1, t2, t3, t4);
+                }
+            }
+        }
+
+        final class Secp256R1GeneratorMultiplier implements PointMultiplier {
+            private static final ECPoint point =
+                    CurveDB.lookup("secp256r1").getGenerator();
+            private static final PointMultiplier multiplier =
+                    new Secp256R1GeneratorMultiplier();
+
+            private static final ImmutableIntegerModuloP zero =
+                    IntegerPolynomialP256.ONE.get0();
+            private static final ImmutableIntegerModuloP one =
+                    IntegerPolynomialP256.ONE.get1();
+
+            @Override
+            public ProjectivePoint.Mutable pointMultiply(byte[] s) {
+                MutableIntegerModuloP t0 = zero.mutable();
+                MutableIntegerModuloP t1 = zero.mutable();
+                MutableIntegerModuloP t2 = zero.mutable();
+                MutableIntegerModuloP t3 = zero.mutable();
+                MutableIntegerModuloP t4 = zero.mutable();
+
+                ProjectivePoint.Mutable d = new ProjectivePoint.Mutable(
+                        zero.mutable(),
+                        one.mutable(),
+                        zero.mutable());
+                ProjectivePoint.Mutable r = d.mutable();
+                for (int i = 15; i >= 0; i--) {
+                    secp256r1Ops.setDouble(d, t0, t1, t2, t3, t4);
+                    for (int j = 3; j >= 0; j--) {
+                        int pos = i + j * 16;
+                        int index = (posit(s, pos + 192) << 3) |
+                                    (posit(s, pos + 128) << 2) |
+                                    (posit(s, pos +  64) << 1) |
+                                     posit(s, pos);
+
+                        lookup(P256.points[j], index, r);
+                        secp256r1Ops.setSum(d, r, t0, t1, t2, t3, t4);
+                    }
+                }
+
+                return d;
+            }
+
+            private static int posit(byte[] k, int i) {
+                return (k[i >> 3] >> (i & 0x07)) & 0x01;
+            }
+
+            // Lazy loading of the tables.
+            private static final class P256 {
+                // Pre-computed tables to speed up the point multiplication.
+                //
+                // This is a 4 dimensions table, and each dimension contains
+                // 16 elements. For the 1st demension, each element in it is
```

Comment:
s/demension/dimension/

Reply:
Thank you for the catch.  Updated in the latest patch.